java/maven/src/org/netbeans/modules/maven/NbMavenProjectImpl.java

```diff
@@ -552,6 +555,50 @@ public RequestProcessor.Task fireProjectReload() {
         }
         return reloadTask;
     }
+    
+    private void reloadPossibleBrokenModules(MavenProject preceding, MavenProject p) {
+        // restrict to just poms that were marked as broken/incomplete.
+        if (!(MavenProjectCache.isFallbackproject(preceding) || 
+            preceding.getContextValue("org.netbeans.modules.maven.problems.primingNotDone") != Boolean.TRUE)) {
+            return;
+        }
+        // but do not cascade from projects, which are themselves broken.
+        if (MavenProjectCache.isFallbackproject(p)) {
+            return;
+        }
+        File basePOMFile = p.getFile().getParentFile();
+        for (String modName : p.getModules()) {
+            File modPom = new File(new File(basePOMFile, modName), "pom.xml");
+            if (!modPom.exists() && modPom.isFile()) {
+                LOG.log(Level.FINE, "POM file {0} for module {1} does not exist", new Object[] { modPom, modName });
+                continue;
+            }
+            MavenProject child = MavenProjectCache.getMavenProject(modPom, true, false);
+            // the project may have more problems, more subtle, but now repair just total breakage
+            if (!MavenProjectCache.isFallbackproject(child)) {
```

Comment:
A possible NPE when `child` obtained from `MavenProjectCache.getMavenProject(modPom, true, false)` is `null` ?

Reply:
Thanks! Fixed in e0159d8f0f86